### ISSUE TITLE
simdata module

### ISSUE THREAD

**Issue by AugustT** [zoonproject/zoon#274]
We need a module that can create simulated data. This will need a number of features (add any I have missed):

- [ ] Ability to specify the extent
- [ ] Specify the number of data points desired
- [ ] Specify the type of data points (presence, absence, background, proportion, ...)
- [ ] Specify how these should be distributed (supply a probability surface?)

**Comment by timcdlucas** [comment]
Probably want someone a bit more generic than a surface. Perhaps it should take a function somehow. 

I can imagine you might want to random select a data point then add datapoints close to the original point. This would require a function rather than a probability surface.

**Comment by goldingn** [comment]
Isn't @RTBecard developing modules like this?

**Comment by RTbecard** [comment]
Yes, that's right.  I'm using simulated datasets to test detection models within zoon, although I haven't incorporated it into a module yet but that should be quite easy to do.

Right now I have a set of functions that do the following:
- Create a random raster of covariates for use in zoon within a specified extent.
- Create site abundances and detections for a specified number of randomly located sites in the raster which vary according to the covariates at that location.
 - I have my site abundances distributed via poisson distribution with a log-linked relationship to covairates.  I've also added support for providing empiracle distributions, so I can test models on datasets which violate the prior distribution assumptions.

I'm using this to check how the number of sites and visits affect the performance of the detection models I'm working with, in addition to using it to make sure my models have been implemented correctly.

If this sounds useful, I can post a vignette over the weekend, and if the functions I'm using are similar to what you guys have in mind for the simdata module, I can adapt it into a module.
